code/controllers/subsystem/processing/overlays.dm

```diff
@@ -0,0 +1,115 @@
+var/datum/subsystem/processing/overlays/SSoverlays
+
+/datum/subsystem/processing/overlays
+	name = "Overlays"
+	flags = SS_TICKER|SS_NO_INIT|SS_FIRE_IN_LOBBY|SS_KEEP_TIMING
+	wait = 1
+	priority = 500
+	can_fire = FALSE
+
+	stat_tag = "Ov"
+	currentrun = null
+
+/datum/subsystem/processing/overlays/New()
+	NEW_SS_GLOBAL(SSoverlays)
+
+/datum/subsystem/processing/overlays/Recover()
+	processing = SSoverlays.processing
+
+/datum/subsystem/processing/overlays/fire()
+	while(processing.len)
+		var/atom/thing = processing[processing.len]
+		processing.len--
+		if(thing)
+			if(LAZYLEN(thing.priority_overlays) && LAZYLEN(thing.our_overlays))
+				thing.overlays = (thing.our_overlays | thing.priority_overlays)
+			else if(LAZYLEN(thing.our_overlays))
+				thing.overlays = thing.our_overlays.Copy()
+			else if(LAZYLEN(thing.priority_overlays))			//Do these two have to be copied? We don't want assignments to them to trigger overlay updates. But, idk, bruh, BYOND. - Cyberboss
+				thing.overlays = thing.priority_overlays.Copy()
+			else
+				LAZYCLEARLIST(thing.overlays)
+		if(MC_TICK_CHECK)
```

Comment:
the issue is how shittily the mc treats subsystems that don't mc_tick_check (delaying until either it there is enough free tick to run all the way or waiting for it to go so long it says fuck it and runs it anyway (it temp increases the priority by 10% every tick until it's first in the queue, then runs it anyways), but if you don't give the flag that tells it the subsystem doesn't mc_tick_check it would be ok.